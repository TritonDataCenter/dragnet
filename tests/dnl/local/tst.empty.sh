#!/bin/bash

#
# tst.empty.sh: tests basic "scan" operations on empty files
#

set -o errexit
. $(dirname $0)/../common.sh

tmpfile="/var/tmp/$(basename $0).$$"
echo "using tmpfile \"$tmpfile\"" >&2

function scan-file
{
	echo "# dnl scan-file" "$@"
	dnl scan-file "$@" /dev/null 2>&1
	echo

	echo "# dnl scan-file --points" "$@"
	dnl scan-file --points "$@" /dev/null 2>&1 | sort -d
	echo
}

function query
{
	echo "# dnl query-file" "$@"
	dnl query-file "$@" $tmpfile 2>&1
}

scan-file --counters
scan-file -b timestamp
scan-file -b timestamp[aggr=quantize]
scan-file -b timestamp[aggr=quantize],req.method
scan-file -f '{ "eq": [ "audit", true ] }' -b timestamp[aggr=quantize],req.method
scan-file --counters -f '{ "eq": [ "audit", true ] }'

echo "creating index" >&2
dnl index-file /dev/null $tmpfile
query --counters

echo "creating index" >&2
dnl index-file -c req.method,latency[aggr=quantize] /dev/null $tmpfile
query --counters
query -f '{ "eq": [ "req.method", "GET" ] }'
query -b req.method
query -b latency
query --counters -b latency
