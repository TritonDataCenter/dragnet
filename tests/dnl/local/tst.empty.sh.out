# dnl scan-file --counters
VALUE
    0
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dnl scan-file --points --counters
Aggregator         noutputs:           1
{"fields":{},"value":0}

# dnl scan-file -b timestamp

# dnl scan-file --points -b timestamp

# dnl scan-file -b timestamp[aggr=quantize]

# dnl scan-file --points -b timestamp[aggr=quantize]

# dnl scan-file -b timestamp[aggr=quantize],req.method

# dnl scan-file --points -b timestamp[aggr=quantize],req.method

# dnl scan-file -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dnl scan-file --points -f { "eq": [ "audit", true ] } -b timestamp[aggr=quantize],req.method

# dnl scan-file --counters -f { "eq": [ "audit", true ] }
VALUE
    0
Aggregator         noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1

# dnl scan-file --points --counters -f { "eq": [ "audit", true ] }
Aggregator         noutputs:           1
{"fields":{},"value":0}

# dnl query-file --counters
VALUE
    0
Index Aggregator   ninputs:            1
Index Aggregator   noutputs:           1
Query Passthrough  ninputs:            1
Query Passthrough  noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dnl query-file --counters
VALUE
    0
Index Aggregator   ninputs:            1
Index Aggregator   noutputs:           1
Query Passthrough  ninputs:            1
Query Passthrough  noutputs:           1
Flattener          ninputs:            1
Flattener          noutputs:           1
# dnl query-file -f { "eq": [ "req.method", "GET" ] }
VALUE
    0
# dnl query-file -b req.method
# dnl query-file -b latency
# dnl query-file --counters -b latency
Flattener          noutputs:           1
